R/length_test_linter.R

```diff
@@ -0,0 +1,55 @@
+#' Check for a common mistake where length is applied in the wrong place
+#'
+#' Usage like `length(x == 0)` is a mistake. If checking the length of `x`,
+#'   `length(x) == 0` was intended. The length of the logical vector `x == 0`
+#'   will be the same as the length of `x`, so it would be more direct to use
+#'   `length(x)` if that value was really intended. Even in more complicated
+#'   cases like `length(x == y)` where `x` and `y` may have different lengths
+#'   and this expression relies on recycling rules, it is far preferable to
+#'   make the intent clearer, e.g. `max(length(x), length(y))`.
```

Comment:
I found this paragraph difficult to read, and I was trying to slightly rephrase it. This is what I came up with. What do you think?

> It's almost always a mistake to use `length(x == 0)`. If you intended to check the length of `x` and see if it's equal to zero, then you should use `length(x) == 0`. Using `x == 0` as a logical vector will give you the same length as `x`, so it's more direct to use `length(x)` if that's what you intended. In more complex cases like `length(x == y)`, where `x` and `y` may have different lengths and rely on recycling rules, it's better to make your intent clearer. For example, you could use `max(length(x), length(y))`.

Reply:
I just axed most of this. Those cases are quite abstruse -- the point is `length(<obj> <comparator> <obj>)` is always a mistake.

Reply:
Much better. Thanks!